### ISSUE TITLE
Mobile version is blank in mSupply on the fresh install

### ISSUE THREAD

**Issue by Mai21** [openmsupply/mobile#5006]
## Describe the bug

There are many mobile sites that don't have version information on the desktop.
In the desktop, the mobile version is populated from message.toVersionString  with type = 'mobile_upgrade'

As discussed with @sworup  when the mobile is newly installed, no message seems to send to the desktop. Only when the mobile is upgraded, the message is sent.

### To reproduce

1. Do a fresh install of mSupply mobile and enter the sync username password to sync data
2. At this point the mobile app should have a site set.
3. In mSupply desktop go to site table to find the site of the current mobile app
4. See current mobile app version is empty

![SCR-20221116-ebo](https://user-images.githubusercontent.com/683452/202127435-f0bbbfc6-8ffb-489a-a9de-64f6fe7c9bd3.png)

### Expected behavior

Even during a fresh install, the desktop site table record for the mobile app's site should have the `version` field populated with the current version of the mobile app.

### Proposed Solution

We have to do two things:

- During fresh install, the mobile version should be sent through message to desktop to update `site.version` properly.
- For fresh install too, add a ` createRecord(database, 'UpgradeMessage', fromVersion, toVersion);` call where fromVersion would be an empty string and toVersion would be the current version. This will tell the dashboard user on which day a fresh installation was done, along with showing the current version.

At this part of the code:
https://github.com/openmsupply/mobile/blob/9ecf8683e74d9cad797191f44c500ca3cd6f5fe6/src/dataMigration.js#L35-L53

Make sure the line below is not triggering a return. 

https://github.com/openmsupply/mobile/blob/9ecf8683e74d9cad797191f44c500ca3cd6f5fe6/src/dataMigration.js#L39

If it is triggering a return, then the update of the Setting would not trigger, hence not triggering the message that would update `site.version` on desktop. 

https://github.com/openmsupply/mobile/blob/9ecf8683e74d9cad797191f44c500ca3cd6f5fe6/src/dataMigration.js#L54-L56

This is where I would start.

### Version and device info

- App version: 8.6.1
- Tablet model:
- OS version:

**Comment by sworup** [comment]
@Mai21 is this good to pick and start working on? I think @arjunSussol is working on another issue but it would be nice to know if this issue could be picked.

**Comment by Mai21** [comment]
@sworup @arjunSussol Yes, please! Thanks!

**Comment by arjunSussol** [comment]
Hi @Mai21 and @sworup today I got chance to look at this issue.
My observations:

1) This is not mobile issue
2) This issue occurs due to mSupply desktop older version i.e. mSupply master(v6).
3) In mSupply develop(v7) we can't notice this issue

Reasons: This is fixed in develop, could be Sync v5 fixed that issue. I noticed new method `REST_authenticateSite` in mSupply develop which handle most of the site header including app-version(mobile-version) header.

So, in my opinion better to wait new version(v7) of mSupply desktop to be released. 
No need to put more effort to mobile and older mSupply version to fix this as this would take more time than we expected and in future the the code changes will create conflict with new version.

However, I am waiting for opinion from you both and others also.
